tests/e2e/utils/service_utils.go

```diff
@@ -100,20 +102,29 @@ func WaitServiceExposureAndGetIPs(cs clientset.Interface, namespace string, name
 }
 
 // WaitServiceExposureAndValidateConnectivity returns IPs of the service and check the connectivity if they are public IPs.
+// Service should have been created before calling this function.
 func WaitServiceExposureAndValidateConnectivity(cs clientset.Interface, ipFamily IPFamily, namespace string, name string, targetIPs []string) ([]string, error) {
 	var service *v1.Service
 	var err error
 	var ips []string
 
+	deleteService := func() {
```

Comment:
You can also make deleteService is a deferred function. When err is not nil, delete the service.

Reply:
Thank you. Updated.